mapie/futur/split/base.py

```diff
@@ -0,0 +1,600 @@
+    def __init__(
+        self,
+        predictor: Optional[
+            Union[
+                Union[RegressorMixin, ClassifierMixin],
+                Pipeline,
+                List[Union[Union[RegressorMixin, ClassifierMixin], Pipeline]]
+            ]
+        ] = None,
+        calibrator: Optional[CCPCalibrator] = None,
+        cv: Optional[
+            Union[str, BaseCrossValidator, BaseShuffleSplit]
+        ] = None,
+        alpha: Optional[float] = None,
+        conformity_score: Optional[ConformityScore] = None,
+        random_state: Optional[int] = None,
+    ) -> None:
+        """
+        Initialisation
+        """
+
+    @abstractmethod
+    def _check_fit_parameters(self) -> Union[RegressorMixin, ClassifierMixin]:
+        """
+        Check and replace default value of ``predictor`` and ``cv`` arguments.
+        """
+
+    @abstractmethod
+    def _check_calibrate_parameters(self) -> Tuple[
+        ConformityScore, BaseCalibrator
+    ]:
+        """
+        Check and replace default ``conformity_score``, ``alpha`` and
+        ``calibrator`` arguments.
+        """
+
+    def _check_cv(
+        self,
+        cv: Optional[Union[str, BaseCrossValidator, BaseShuffleSplit]] = None,
+        test_size: Optional[Union[int, float]] = None,
+    ) -> Union[str, BaseCrossValidator, BaseShuffleSplit]:
+        """
+        Check if ``cv`` is ``None``, ``"prefit"``, ``"split"``,
+        or ``BaseShuffleSplit``/``BaseCrossValidator`` with ``n_splits``=1.
+        Return a ``ShuffleSplit`` instance ``n_splits``=1
+        if ``None`` or ``"split"``.
+        Else raise error.
+
+        Parameters
+        ----------
+        cv: Optional[Union[str, BaseCrossValidator, BaseShuffleSplit]]
+            Cross-validator to check, by default ``None``.
+
+        test_size: float
+            If float, should be between 0.0 and 1.0 and represent the
+            proportion of the dataset to include in the test split.
+            If cv is not ``"split"``, ``test_size`` is ignored.
+
+            By default ``None``.
+
+        Returns
+        -------
+        Union[str, PredefinedSplit, ShuffleSplit]
+            The cast `cv` parameter.
+
+        Raises
+        ------
+        ValueError
+            If the cross-validator is not valid.
+        """
+        if cv is None or cv == "split":
+            return ShuffleSplit(
+                n_splits=1, test_size=test_size, random_state=self.random_state
+            )
+        elif (isinstance(cv, (PredefinedSplit, ShuffleSplit))
+              and cv.get_n_splits() == 1):
+            return cv
+        elif cv == "prefit":
+            return cv
+        else:
+            raise ValueError(
+                "Invalid cv argument.  Allowed values are None, 'prefit', "
+                "'split' or a ShuffleSplit/PredefinedSplit object with "
+                "``n_splits=1``."
+            )
+
+    def _check_alpha(self, alpha: Optional[float] = None) -> None:
```

Comment:
Is this not a generic function?

Reply:
It is generic more ``'split'`` only methods. But we don't have any right now. (Except for CQR but the check_alpha returns (0.5, alpha/2, 1-alpha/2), which we don't want here)